encode/src/main/scala/org/broadinstitute/monster/etl/encode/EncodeTransforms.scala

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.monster.etl.encode
+
+import com.spotify.scio.values.SCollection
+import io.circe.{Json, JsonObject}
+import io.circe.syntax._
+
+object EncodeTransforms {
+  type JsonPipe = SCollection[JsonObject] => SCollection[JsonObject]
+
+  def cleanEntities(entityType: EncodeEntity): JsonPipe = { stream =>
+    val name = entityType.entryName
+    stream
+      .transform(s"Trim $name Fields")(trimFields(entityType.fieldsToKeep))
+      .transform(s"Rename $name Fields")(renameFields(entityType.fieldsToRename))
+      .transform(s"Build $name Links")(buildLinks(entityType.linkFields))
+      .transform(s"Extract $name Labels")(extractLabels)
+      .transform(s"Combine $name Aliases")(collectAliases(entityType.aliasFields))
+  }
+
+  private val extractLabels: JsonPipe = { stream =>
+    val idRegex = "/[^/]+/(.+)/".r
+
+    stream.map { json =>
+      val extracted = for {
+        idJson <- json("label")
+        idString <- idJson.asString
+        label <- idRegex.findFirstMatchIn(idString)
+      } yield {
+        label
+      }
+
+      extracted
+        .fold(json) { labelMatch =>
+          val label = labelMatch.group(1)
+          json
+            .add("label", label.asJson)
+            .add("id", s"Broad-$label".asJson)
+        }
+    }
+  }
+
+  private def trimFields(fieldsToKeep: Set[String]): JsonPipe =
+    _.map(_.filterKeys(fieldsToKeep))
+
+  private def renameFields(fieldsToRename: Map[String, String]): JsonPipe =
+    _.map { json =>
+      fieldsToRename.foldLeft(json) {
+        case (renamedSoFar, (oldName, newName)) =>
+          renamedSoFar(oldName).fold(renamedSoFar) { value =>
+            renamedSoFar.add(newName, value).remove(oldName)
+          }
+      }
+    }
+
+  private def buildLinks(linkFields: Set[String]): JsonPipe =
+    _.map { json =>
+      linkFields.foldLeft(json) { (linkedSoFar, fieldName) =>
+        val linkValue = for {
+          valueJson <- linkedSoFar(fieldName)
+          valueString <- valueJson.asString
+        } yield {
+          if (valueString.charAt(0) == '/') {
```

Comment:
This looks great for a first past, but maybe a check to make sure it can handle any edge cases could possibly be helpful